File: server/todolistJava01/src/main/java/jp/co/linnefromice/todolist01/gateway/restapi/TaskRestApiController.java
package jp.co.linnefromice.todolist01.gateway.restapi;

import jp.co.linnefromice.todolist01.model.Task;
import jp.co.linnefromice.todolist01.service.TaskService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/gateway/restapi/tasks")
public class TaskRestApiController {

    @Autowired
    private TaskService taskService;

    @GetMapping(path="findAll")
    public List<Task> findAll() {
        return taskService.findAll();
    }
}

File: server/todolistKotlin01/src/main/kotlin/jp/co/linnefromice/todolistKotlin01/gateway/restapi/TaskRestApiController.kt
package jp.co.linnefromice.todolistKotlin01.gateway.restapi

import jp.co.linnefromice.todolistKotlin01.model.Task
import jp.co.linnefromice.todolistKotlin01.service.TaskService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/gateway/restapi/tasks")
class TaskRestApiController(
    val service: TaskService) {

    @GetMapping(path= ["findAll"])
    fun findAll(): List<Task> {
        return service.findAll();
    }
}